src/hotspot/cpu/x86/x86.ad

```diff
@@ -7866,121 +7865,110 @@ instruct loadMask(legVec dst, legVec src) %{
   %}
   ins_pipe( pipe_slow );
 %}
 
-
-instruct loadMask_evex(kReg dst, vec src, rRegP scratch, kReg kscratch) %{
+instruct loadMask_evex(kReg dst, vec src,  vec xtmp) %{
   predicate(n->bottom_type()->isa_vectmask() && VM_Version::supports_avx512vlbw());
   match(Set dst (VectorLoadMask src));
-  effect(TEMP scratch, TEMP kscratch);
-  format %{ "vector_loadmask_byte $dst,$src\n\t" %}
+  effect(TEMP xtmp);
+  format %{ "vector_loadmask_byte $dst, $src\t! using $xtmp as TEMP" %}
   ins_encode %{
     int vlen_enc = vector_length_encoding(in(1));
-    __ evpcmp(T_BYTE, $dst$$KRegister, k0, $src$$XMMRegister, ExternalAddress(vector_mask_cmp_bits()),
-              Assembler::eq, vlen_enc, $scratch$$Register);
+    __ load_vector_mask($dst$$KRegister, $src$$XMMRegister, $xtmp$$XMMRegister,
+                        noreg, false, vlen_enc);
   %}
   ins_pipe( pipe_slow );
 %}
 
 //------------------------------------- StoreMask --------------------------------------------
 
-instruct storeMask1B(vec dst, vec src, immI_1 size) %{
-  predicate((Matcher::vector_length(n) < 64 || VM_Version::supports_avx512vlbw()) &&
-            n->in(1)->bottom_type()->isa_vectmask() == NULL);
+instruct vstoreMask1B(vec dst, vec src, immI_1 size) %{
+  predicate(Matcher::vector_length(n) < 64 && n->in(1)->bottom_type()->isa_vectmask() == NULL);
   match(Set dst (VectorStoreMask src size));
-  format %{ "vector_store_mask $dst,$src\t!" %}
+  format %{ "vector_store_mask $dst, $src \t! elem size is $size byte[s]" %}
   ins_encode %{
-    assert(UseSSE >= 3, "required");
-    if (Matcher::vector_length_in_bytes(this) <= 16) {
+    int vlen = Matcher::vector_length(this);
+    if (vlen <= 16 && UseAVX <= 2) {
+      assert(UseSSE >= 3, "required");
       __ pabsb($dst$$XMMRegister, $src$$XMMRegister);
     } else {
-      assert(UseAVX >= 2, "required");
+      assert(UseAVX > 0, "required");
       int src_vlen_enc = vector_length_encoding(this, $src);
       __ vpabsb($dst$$XMMRegister, $src$$XMMRegister, src_vlen_enc);
     }
   %}
   ins_pipe( pipe_slow );
 %}
 
-instruct storeMask2B(vec dst, vec src, immI_2 size) %{
-  predicate(Matcher::vector_length(n) <= 8 &&
-            n->in(1)->bottom_type()->isa_vectmask() == NULL);
-  match(Set dst (VectorStoreMask src size));
-  format %{ "vector_store_mask $dst,$src\n\t" %}
-  ins_encode %{
-    assert(UseSSE >= 3, "required");
-    __ pabsw($dst$$XMMRegister, $src$$XMMRegister);
-    __ packsswb($dst$$XMMRegister, $dst$$XMMRegister);
-  %}
-  ins_pipe( pipe_slow );
-%}
-
 instruct vstoreMask2B(vec dst, vec src, immI_2 size) %{
-  predicate(Matcher::vector_length(n) == 16 && !VM_Version::supports_avx512bw() &&
-            n->in(1)->bottom_type()->isa_vectmask() == NULL);
+  predicate(Matcher::vector_length(n) <= 16 && n->in(1)->bottom_type()->isa_vectmask() == NULL);
   match(Set dst (VectorStoreMask src size));
-  effect(TEMP dst);
-  format %{ "vector_store_mask $dst,$src\t!" %}
+  effect(TEMP_DEF dst);
```

Comment:
The original storeMask2B didn't need TEMP dst and so was a separate rule.

Reply:
TEMP_DST ensures creation of separate machine operands which gets allocated same register as DST, thus preventing any accidental sharing of register with any other temporary machine operand over the node.  In this case since there is no other explicit temporary register used to effectively, TEMP_DEF  is same at DEF for all practical purpose.